src/common/utxobased/engine/makeUtxoEngineState.ts

```diff
@@ -63,13 +63,22 @@ export function makeUtxoEngineState(config: UtxoEngineStateConfig): UtxoEngineSt
 
   return {
     async start(): Promise<void> {
+      let processedCount = 0
```

Comment:
You need to keep the 'ratio' value so you are able to throttle emitting the event.

Check this line in the old implementation for more detail:
https://github.com/EdgeApp/edge-currency-bitcoin/blob/698f4609a11ae8fd88ff3a8b53b1b3de40c1d0ec/src/engine/engineState.js#L511

Reply:
So the event is only emitted at max 4 times for each wallet?